### PR TITLE
[Shop][Checkout] Seeing shipping and payment methods on order summary

### PR DESCRIPTION
**Pull request by Arminek**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| License         | MIT

### CODE REVIEW

File: features/checkout/seeing_order_summary/seeing_shipping_method_and_payment_method_on_order_summary_page.feature [hunk 1]
```diff
@@ -0,0 +1,21 @@
+@checkout
+Feature: Seeing order shipping method details on summary page
+    In order to be certain about shipping method
```
Comment (tuka217) [inline features/checkout/seeing_order_summary/seeing_shipping_method_and_payment_method_on_order_summary_page.feature]: about **a** shipping method?

File: src/Sylius/Behat/Page/Shop/Checkout/SummaryPage.php [hunk 2]
```diff
@@ -51,11 +53,45 @@ public function hasBillingAddress(AddressInterface $address)
     /**
      * {@inheritdoc}
      */
+    public function hasShippingMethod(ShippingMethodInterface $shippingMethod)
+    {
+        if (!$this->hasElement('shipping_method')) {
+            return false;
+        }
+
+        if (false !== strpos($this->getElement('shipping_method')->getText(), $shippingMethod->getName())) {
```
Comment (tuka217) [inline src/Sylius/Behat/Page/Shop/Checkout/SummaryPage.php]: `return false !== strpos(..)`, WDYT?

File: src/Sylius/Behat/Page/Shop/Checkout/SummaryPage.php [hunk 3]
```diff
@@ -51,11 +53,45 @@ public function hasBillingAddress(AddressInterface $address)
     /**
      * {@inheritdoc}
      */
+    public function hasShippingMethod(ShippingMethodInterface $shippingMethod)
+    {
+        if (!$this->hasElement('shipping_method')) {
```
Comment (tuka217) [inline src/Sylius/Behat/Page/Shop/Checkout/SummaryPage.php]: 👍

File: src/Sylius/Behat/Page/Shop/Checkout/SummaryPage.php [hunk 4]
```diff
@@ -51,11 +53,45 @@ public function hasBillingAddress(AddressInterface $address)
...
+    /**
+     * {@inheritdoc}
+     */
+    public function hasPaymentMethod(PaymentMethodInterface $paymentMethod)
+    {
+        if (!$this->hasElement('payment_method')) {
+            return false;
+        }
+
+        if (false !== strpos($this->getElement('payment_method')->getText(), $paymentMethod->getName())) {
```
Comment (tuka217) [inline src/Sylius/Behat/Page/Shop/Checkout/SummaryPage.php]: Same here

File: features/checkout/seeing_order_summary/seeing_shipping_method_and_payment_method_on_order_summary_page.feature [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        And I am a logged in customer
+
+    @ui
+    Scenario: Seeing shipping method and payment method
+        Given I have product "Lannister Coat" in the cart
+        When I specified the shipping address as "Ankh Morpork", "Frost Alley", "90210", "France" for "Jon Snow"
+        And I proceed order with "Cash on delivery" shipping method and "Offline" payment
+        Then I should be on the checkout summary step
+        And I should see "Cash on delivery" shipping method
+        And I should see "offline" payment method
```
Comment (michalmarcinkowski) [inline features/checkout/seeing_order_summary/seeing_shipping_method_and_payment_method_on_order_summary_page.feature]: And my order's shipping method should be "Cash on delivery"
And my order's payment method should be "offline"

File: features/checkout/seeing_order_summary/seeing_shipping_total_on_order_summary_page.feature [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        And the store allows paying offline
+        And I am a logged in customer
+
+    @ui
+    Scenario: Seeing the shipping total on order summary
+        Given I have product "The Sorting Hat" in the cart
+        When I specified the shipping address as "Ankh Morpork", "Frost Alley", "90210", "France" for "Jon Snow"
+        And I proceed order with "UPS" shipping method and "Offline" payment
+        Then I should be on the checkout summary step
+        And I should see the shipping total with price "€20.00"
```
Comment (michalmarcinkowski) [inline features/checkout/seeing_order_summary/seeing_shipping_total_on_order_summary_page.feature]: And my order shipping should be "$20"

File: features/checkout/seeing_order_summary/seeing_total_discount_on_order_summary_page.feature [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+    @ui
+    Scenario: Seeing the total discount on order summary page
+        Given I have product "The Sorting Hat" in the cart
+        When I specified the shipping address as "Ankh Morpork", "Frost Alley", "90210", "France" for "Jon Snow"
+        And I proceed order with "Free" shipping method and "Offline" payment
+        Then I should be on the checkout summary step
+        And I should see promotion total "-€9.00"
+        And I should see promotion "Holiday promotion"
+        And I should see promotion "All year promotion"
```
Comment (michalmarcinkowski) [inline features/checkout/seeing_order_summary/seeing_total_discount_on_order_summary_page.feature]: And my order promotion total should be "-$9.00"
And "Holiday promotion" should be applied to my order
And "All year promotion" should be applied to my order

File: src/Sylius/Behat/Context/Setup/PaymentContext.php [hunk 8]
```diff
@@ -87,6 +87,7 @@ public function storeAllowsPaying($paymentMethodName)
         $channel = $this->sharedStorage->get('channel');
         $channel->addPaymentMethod($paymentMethod);
 
+        $this->sharedStorage->set('payment_method', $paymentMethod);
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Setup/PaymentContext.php]: Do we need this?

File: src/Sylius/Bundle/ShopBundle/Resources/views/Checkout/summary.html.twig [hunk 9]
```diff
@@ -30,17 +30,22 @@
...
                 </tr>
             {% endfor %}
+            </tbody>
+            <tfoot>
             <tr>
-                <td colspan="4" class="right aligned" id="tax-total">
-                    {{ 'sylius.ui.tax_total'|trans}} : {{ order.taxTotal|sylius_price(order.currencyCode) }}
-                </td>
+                <th colspan="4" class="right aligned" id="tax-total">
+                    {{ 'sylius.ui.tax'|trans }}: {{ order.taxTotal|sylius_price }}
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/ShopBundle/Resources/views/Checkout/summary.html.twig]: Should be `|sylius_price(order.currencyCode)`

**Comment by pjedrzejewski** [comment]
Thank you Arek, Ania and Mateusz! :)

**Pull request merged**